classes/class11/README.md

```diff
@@ -1 +1,187 @@
+
+###Avoid Script Injection With Input Sanitization
+The easiest way to prevent a script injection from happening in your application is to validate all inputs and catch any potential attacks.
+There are [plenty](https://www.npmjs.com/search?q=sanitization) of npm [libraries](https://www.npmjs.com/search?q=sanitize) for both front end validation (this field expects a number, don't submit unless it is just a number), as well as back end sanitization (escape html tags and other special characters like `\/?(){}`). Check out OWASP's [super detailed rules](https://www.owasp.org/index.php/XSS_%28Cross_Site_Scripting%29_Prevention_Cheat_Sheet) for preventing XSS generally.
+
+###Don't leak stack traces
+Now, I'm going to preface this section by telling you a truth that most people struggle with and usually end up paying dearly for misunderstanding:
+**[security through obscurity is a myth](http://en.wikipedia.org/wiki/Security_through_obscurity)**.
+That is to say, your codebase can (and should) be entirely open source, and still be secure.
+An attacker should not gain any benefit from knowing how your application is implemented.
+If you ever think to your self, "hmm, this should be safe as long as no one ever finds out about this, and how could they," you're probably suffering from the delusion of security through obscurity.
+
+So, with that said - assume our codebase and system details (software/library versions) are secret.
+How can an attacker find vulnerabilities in your system?
+By causing errors of course!
+And if you're not careful about catching errors, they might get sent to the client containing information you might not want the user to see, including stack traces.
+Stack traces can tell an attacker not only what your code is trying to do, but also what version of a particular library you are using, or even what version of node. If they know vulnerabilities in those systems, well now they know vulnerabilities in your system.
+
+The standard express boilerplate uses the following code after all other routes to catch any internal errors and prevent them from showing up in a production environment.
+This will catch any unexpected errors in express, but it's still up to you to catch any errors within your own routes and not send entire stack traces.
+```js
+// development error handler
+// will print stacktrace
+if (app.get('env') == 'development') {
+  app.use(function(err, req, res, next) {
+      res.status(err.status || 500);
+      res.render('error', {
+          message: err.message,
+          error: err
+      });
+  });
+}
+
+// production error handler
+// no stacktraces leaked to user
+app.use(function(err, req, res, next) {
+  res.status(err.status || 500);
+  res.render('error', {
+    message: err.message,
+    error: {}
```

Comment:
Maybe a comment on this line to highlight the difference? I almost thought this was wrong because it was such a minor  change.